lib/src/config_handler.dart

```diff
@@ -131,7 +132,7 @@ class ConfigHandler {
     configFile.writeAsStringSync(newConfigString);
 
     // Update the [ToolInfo] object for the current tool
-    parsedTools[tool]!.lastRun = DateTime.now();
+    parsedTools[tool]!.lastRun = clock.now();
```

Comment:
You can avoid the `!` operator if you rewrite the first statement in this function as:

```dart
final ToolInfo? toolInfo = parsedTools[tool];
if (toolInfo == null) {
  return;
}
```

And then these lines can be (via type promotion):

```dart
    // Update the [ToolInfo] object for the current tool
    toolInfo.lastRun = clock.now();
    toolInfo.versionNumber = newVersionNumber;
```